lib/alex_koin/slack_commands.ex

```diff
@@ -80,8 +80,39 @@ defmodule AlexKoin.SlackCommands do
     # 1. get coins to transfer
     coins = Repo.all(Coin.for_wallet(from_wallet, amount))
 
-    # 2. move the coins over to the other wallet
-    Enum.each(coins, fn(c) -> transfer_coin(c, from_wallet, to_wallet, memo) end)
+    # 2. Create the transaction, for now use the first coin's id
```

Comment:
Consider rewriting as this

Multi FTW!

```ex
def transfer(%Account{id: from_wallet_id}, %Account{id: to_wallet_id}, amount, memo) do
  Multi.new
  |> Multi.run(:coins, fn ->
    from_wallet
    |> Coin.for_wallet(amount)
    |> Repo.all
  end)
  |> Multi.run(:transaction, fn %{coins: [%{id: coin_id} | _]} ->
    %{
      amount: amount,
      memo: memo,
      from_id: from_wallet_id,
      to_id: to_wallet_id,
      coin_id: coin_id
    }
    |> Transaction.changeset()
    |> Repo.insert!()
  end)
  |> Multi.run(:transfer_coin_ownership, fn %{coins: coins} -> 
    coin_ids = Enum.map(coins, &(&1.id))

    from(c in Coin, 
      where: c.id in ^coin_ids,
      where: c.wallet_id == ^from_wallet_id
    )
    |> Repo.all
    |> Enum.each(fn coin ->
      coin
      |> Coin.changeset(%{wallet_id: to_wallet_id})
      |> Repo.update()
    end)
  end)
  |> Multi.run(:decrement_from_wallet, fn ->
    from_wallet = Repo.get(Wallet, from_wallet_id)
    AlexKoin.Account.update_wallet(
      from_wallet, 
      %{balance: from_wallet.balance - amount}
    )
  end)
  |> Multi.run(:increment_to_wallet, fn ->
    to_wallet = Repo.get(Wallet, to_wallet_id)
    AlexKoin.Account.update_wallet(
      to_wallet, 
      %{balance: to_wallet.balance + amount}
    )
  end)
  |> Repo.transaction
  |> case do
    {:ok, %{transaction: transaction}} -> {:ok, transaction}
    {:error, err} -> {:error, err}
  end
end
```

Reply:
Multis are dope!

Would recommend splitting each chunk into its own function for readability.